src/js/page/ui/settings.js

```diff
@@ -109,35 +113,28 @@ export default class Settings {
   getSettings() {
     // fingerprint is used for cache lookups
     const fingerprint = [];
-
     const output = {
-      plugins: {}
+      plugins: {},
     };
 
-    this._globalInputs.forEach(inputEl => {
-      if (inputEl.name != 'gzip' && inputEl.name != 'original') {
-        if (inputEl.type == 'checkbox') {
-          fingerprint.push(Number(inputEl.checked));
-        }
-        else {
-          fingerprint.push('|' + inputEl.value + '|');
-        }
-      }
+    for (const inputEl of this._globalInputs) {
+      if (inputEl.name === 'gzip' && inputEl.name === 'original') continue;
 
-      if (inputEl.type == 'checkbox') {
+      if (inputEl.type === 'checkbox') {
+        fingerprint.push(Number(inputEl.checked));
         output[inputEl.name] = inputEl.checked;
-      }
-      else {
+      } else {
+        fingerprint.push(`|${inputEl.value}|`);
         output[inputEl.name] = inputEl.value;
       }
-    });
+    }
 
-    this._pluginInputs.forEach(inputEl => {
+    for (const inputEl of this._pluginInputs) {
       fingerprint.push(Number(inputEl.checked));
       output.plugins[inputEl.name] = inputEl.checked;
-    });
+    }
 
-    output.fingerprint = fingerprint.join();
+    output.fingerprint = fingerprint.join(',');
```

Comment:
This isn't quite right. The idea is:

- Inputs other than `gzip` and `original` are used to form the fingerprint.
- All inputs values are gathered into `output`.

Right now it's including `gzip` and `original` in the fingerprint. The first `if` is never true.